pkg/execution/plugins/atlassian/iamrole/iam_role_manager.go

```diff
@@ -328,69 +306,49 @@ func makeStatementKey(statement *IamPolicyStatement) (string, error) {
 	return key.String(), nil
 }
 
-func mergePolicies(policies []*IamPolicyDocument) (*IamPolicyDocument, error) {
+func mergePolicies(policies []IamPolicyDocument) (IamPolicyDocument, error) {
 	// This not only puts all statements in one policy, it also merges
 	// statements that have the same actions/conditions.
-	statementsByKey := make(map[string]*IamPolicyStatement)
+	statementsByKey := make(map[string]IamPolicyStatement)
 	for _, policy := range policies {
 		for _, statement := range policy.Statement {
 			statementKey, err := makeStatementKey(&statement)
 			if err != nil {
-				return nil, err
+				return IamPolicyDocument{}, err
 			}
 			for _, resource := range statement.Resource {
-				if _, present := statementsByKey[statementKey]; !present {
-					statementsByKey[statementKey] = &IamPolicyStatement{
+				statementByKey, present := statementsByKey[statementKey]
+				if !present {
+					statementByKey = IamPolicyStatement{
 						Effect:    "Allow",
 						Action:    statement.Action,
 						NotAction: statement.NotAction,
 						Condition: statement.Condition,
 					}
 				}
-				statementsByKey[statementKey].Resource = append(statementsByKey[statementKey].Resource, resource)
+				statementByKey.Resource = append(statementByKey.Resource, resource)
```

Comment:
This is appending a resource to a thing called "Resource". That makes no sense - should be "Resources" if it's a slice/array.

Reply:
Thank amazon. The type reflects JSON structure of iam policy object. `resource` can be either a string or an array in their policy. 🤷‍♂️